### ISSUE TITLE
Wrong render of colormap in pdf

### ISSUE THREAD

**Issue by brunetto** [veusz/veusz#112]
Hi!
I have a colormap with a wide region that should be completely black (zero value).
When I export the colormap to file, svg and png work fine, the pdf instead shows a gradient instead of a solid black.
I can provide both the veusz file and the resulting pdf if you want.

**Comment by jeremysanders** [comment]
It would be useful to have an example to work out what's going wrong

**Comment by jeremysanders** [comment]
Thanks - that's worrying - it looks like Qt is doing something funny to the image. I'm guessing if you upscale the data to have more pixels the problem might go away. It seems to be ok with EPS, but not with PDF.

**Comment by brunetto** [comment]
I'll try!!
I also found the svg visualized correctly by inkscape, but chrome does
apply some smooth over the whole colormap... see the two attached png... it
is the same file... I never had this problem with histograms and scatter
plot... is it possible that the colormap is managed in a strange way?

brunetto

**Comment by jeremysanders** [comment]
The image is sent to Qt as a bitmap. Normal plots are vector objects. I've got the horrible feeling that Qt is JPEG encoding the bitmap sent to the PDF file, which leads to these artefacts as there are very few pixels in the image.

**Comment by ssgromov** [comment]
@jeremysanders 
yes, that is exactly the case. i was about to open an issue when i stumbled over this one here. is it possible to represent a picture (which is a Qt bitmap) with an array of rectangles (e.g. like the xy widget with squares representing pixels) which will be output to PDF/SVG as true vector data?

**Comment by jeremysanders** [comment]
Just to note I've rewritten the Image widget. This can now draw using rectangles, or upscaling an image to increase the DPI. This should fix the problems with colours and smoothing in SVG output.

Please let me know if this helps.

One side effect is that images on log axes, by default the pixels are mapped to these axes.

**Comment by ssgromov** [comment]
> Just to note I've rewritten the Image widget. This can now draw using rectangles, or upscaling an image to increase the DPI. This should fix the problems with colours and smoothing in SVG output.
> 
> Please let me know if this helps.
> 
> One side effect is that images on log axes, by default the pixels are mapped to these axes.

Zillion thanks, Jeremy! 
Is it possible to pick up a Windows snapshot where it is enabled?
I seem to be unable to find any version past 3.1 or any dev. for Windows to download from here...
